pkg/segment/structs/segstructs.go

```diff
@@ -765,6 +752,9 @@ func (ss *SegStats) Merge(other *SegStats) {
 		}
 	}
 
+	UpdateMinMax(ss, other.Min)
+	UpdateMinMax(ss, other.Max)
```

Comment:
nit: there was somewhere else you also called these together. If they're always called together, I think updating the min and max should be done in one function call. But maybe there's cases where we call it just for one of them?

Reply:
Yeah, we do not always need to call both of these always. For example when we would build the SST we would only update these based on the record value that we see and that would be one at a time.